include/deal.II/base/quadrature_point_data.h

```diff
@@ -707,6 +713,14 @@ CellDataStorage<CellIteratorType, DataType>::get_data(
   return res;
 }
 
+template <typename CellIteratorType, typename DataType>
+bool
+CellDataStorage<CellIteratorType, DataType>::stores_data(
+  const CellIteratorType &cell) const
```

Comment:
You might want to mark the function as `inline` since it lives in a header file.

Reply:
All of the function definitions in this file don't explicitly contain the `inline` keyword but they are all implicitly inline because they are templated.

Reply:
That's not correct. `template` functions have weak linkage, but they're not automatically `inline`.

Reply:
Okay! Do you think other function definitions in this header file need `inline` keyword as well? or is it just this `stores_data()` method because it is small?

Reply:
In principle, every function that is small enough that inlining makes sense. That requires inspection of every function defined in the .h file.